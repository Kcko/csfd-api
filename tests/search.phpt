<?php

/**
 * Search
 * @package Mikulas\Csfd
 */

use Tester\Assert;
use Mikulas\Csfd\Csfd;

require __DIR__ . '/bootstrap.php';


$csfd = new Csfd();
$res = $csfd->findMovie('rather');

$expected = unserialize(file_get_contents(__DIR__ . '/search_result.dat'));
Assert::equal($res, $expected);
