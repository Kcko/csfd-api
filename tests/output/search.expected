array(
	Csfd\Movie::__set_state(array(
		'id' => 2260,
		'year' => 1968,
		'poster_url' => 'http://img.csfd.cz/posters/0/2260_1.jpg',
		'names' => array(
			'cs' => 'Ragazza piuttosto complicata, Una',
		),
		'genres' => array(
			'Thriller',
		),
		'countries' => array(
			'Itálie',
		),
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
		'authors' => array(
			'directors' => array(
				Csfd\Author::__set_state(array(
					'id' => 2865,
					'name' => 'Damiano Damiani',
				)),
			),
			'actors' => array(
				Csfd\Author::__set_state(array(
					'id' => 12145,
					'name' => 'Jean Sorel',
				)),
				Csfd\Author::__set_state(array(
					'id' => 58999,
					'name' => 'Catherine Spaak',
				)),
			),
		),
	)),
	Csfd\Movie::__set_state(array(
		'id' => 12571,
		'year' => 1998,
		'poster_url' => 'http://img.csfd.cz/assets/images/poster-free.png',
		'names' => array(
			'cs' => 'Takové anglické manželství',
		),
		'genres' => array(
			'Drama',
		),
		'countries' => array(
			'Velká Británie',
		),
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
		'authors' => array(
			'directors' => array(
				Csfd\Author::__set_state(array(
					'id' => 14848,
					'name' => 'Paul Seed',
				)),
			),
			'actors' => array(
				Csfd\Author::__set_state(array(
					'id' => 278,
					'name' => 'Albert Finney',
				)),
				Csfd\Author::__set_state(array(
					'id' => 14272,
					'name' => 'Tom Courtenay',
				)),
			),
		),
	)),
	Csfd\Movie::__set_state(array(
		'id' => 80066,
		'year' => 1964,
		'poster_url' => 'http://img.csfd.cz/posters/8/80066_1.jpg',
		'names' => array(
			'cs' => 'I\'d Rather Be Rich',
		),
		'genres' => array(
			'Komedie',
		),
		'countries' => array(
			'USA',
		),
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
		'authors' => array(
			'directors' => array(
				Csfd\Author::__set_state(array(
					'id' => 9772,
					'name' => 'Jack Smight',
				)),
			),
			'actors' => array(
				Csfd\Author::__set_state(array(
					'id' => 21062,
					'name' => 'Sandra Dee',
				)),
				Csfd\Author::__set_state(array(
					'id' => 61588,
					'name' => 'Robert Goulet',
				)),
			),
		),
	)),
	Csfd\Movie::__set_state(array(
		'id' => 325859,
		'year' => 1982,
		'poster_url' => 'http://img.csfd.cz/assets/images/poster-free.png',
		'names' => array(
			'cs' => 'I\'d Rather Be in Paris',
		),
		'genres' => array(
			'Krátkometrážní',
		),
		'countries' => array(
			'USA',
		),
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
		'authors' => array(
			'directors' => array(
				Csfd\Author::__set_state(array(
					'id' => 83163,
					'name' => 'Dominic Angerame',
				)),
			),
		),
	)),
	Csfd\Movie::__set_state(array(
		'id' => 238207,
		'year' => 2003,
		'poster_url' => NULL,
		'names' => NULL,
		'genres' => NULL,
		'countries' => NULL,
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
	)),
	Csfd\Movie::__set_state(array(
		'id' => 311185,
		'year' => 2012,
		'poster_url' => NULL,
		'names' => NULL,
		'genres' => NULL,
		'countries' => NULL,
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
	)),
	Csfd\Movie::__set_state(array(
		'id' => 337999,
		'year' => 1971,
		'poster_url' => NULL,
		'names' => NULL,
		'genres' => NULL,
		'countries' => NULL,
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
	)),
	Csfd\Movie::__set_state(array(
		'id' => 283324,
		'year' => 2008,
		'poster_url' => NULL,
		'names' => NULL,
		'genres' => NULL,
		'countries' => NULL,
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
	)),
	Csfd\Movie::__set_state(array(
		'id' => 252742,
		'year' => 2007,
		'poster_url' => NULL,
		'names' => NULL,
		'genres' => NULL,
		'countries' => NULL,
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
	)),
	Csfd\Movie::__set_state(array(
		'id' => 327393,
		'year' => 2010,
		'poster_url' => NULL,
		'names' => NULL,
		'genres' => NULL,
		'countries' => NULL,
		'rating' => NULL,
		'plot' => NULL,
		'content_rating' => NULL,
		'runtime' => NULL,
		'releases' => NULL,
	)),
)